src/components/progress/progressRing.svelte

```diff
@@ -0,0 +1,93 @@
+<svelte:options tag="leo-progressring" />
+
+<script lang="ts">
+  export let progress = 0
+  export let radius = 24
+  export let strokeWidth = 4
+  export let mode: 'determinate' | 'indeterminate' = 'indeterminate'
+
+  $: progress = mode === 'indeterminate' ? 0.25 : progress
+
+  $: normalizedRadius = radius - strokeWidth
+  $: circumference = normalizedRadius * 2 * Math.PI
+</script>
+
+<div
+  class="leo-progressring"
+  class:spin={mode === 'indeterminate'}
+  style="width: {radius * 2}px; height: {radius * 2}px"
+>
+  <slot />
+  <svg width={radius * 2} height={radius * 2}>
```

Comment:
Wondering whether we can set this to a 24x24 viewbox, and always set the radius of the circle to 24, then allow width / height of this component to be changed via css property, similar (or same as) `--leo-icon-size` @fallaciousreasoning